include/query.php

```diff
@@ -105,6 +105,10 @@ public function get_queried_taxonomies() {
 	 * @return void
 	 */
 	public function set_language( $lang ) {
+		if ( ! $this->can_filter_query() ) {
+			return;
+		}
```

Comment:
With this return here, it doesn't seem worth going into the `! $this->is_already_fitlered( $qvars )` condition in `PLL_Query::filter_query()` when `PLL_Query::can_filter_query()` returns `false`.